code-challenges.test.js

```diff
@@ -1,59 +1,163 @@
+    const object2 = { number: 0 }
+    const object3 = { number: -7 }
+    it("checks if a number is evenly divisible by three", () => {
+        expect(cerberus(object1.number)).toEqual(`${object1.number} is divisible by three! :D`)
+        expect(cerberus(object2.number)).toEqual(`${object2.number} is divisible by three! :D`)
+        expect(cerberus(object3.number)).toEqual(`${object3.number} is not divisible by three! >:(`)
+    })
+});
+// [1st error] ReferenceError: divDecider is not defined
+// [2nd error]  Expected: "[object Object] is divisible by three!" Received: undefined  /// ----> didn't give the test any values to work with.
+
+
+
+// Expected output: "15 is divisible by three"
+// Expected output: "0 is divisible by three"
+// Expected output: "-7 is not divisible by three"
+
+// b) Create the function that makes the test pass.
+
+
+const cerberus = (input) => {     // *****I spent infinitely too long refactoring this question in order to stop it working on boolean values and strings, while still accepting either numbers or objects with the 'number:' property...*****
+    let bone;
+    if (typeof input === "number") {                                 //***** here we check if the input is a strict number, and if so, put it into a nice little treat for cerberus*****
+      bone = input;
+    } else if (typeof input === "object" && !isNaN(input.number)) {
+      bone = input.number;                                           // *****DOUBLENEGATIVESAREATHINGINCODEDEARGODWHYMYPOORBRAIN (T﹏T) *ahem* sorry. this bit checks if the given input is of type "object" and is not NOT a number, and if both conditions are true, stuffs the not NOT number into our bone value. gonna be real, I lucked into that answer. could also replace the double negative with the much simpler [typeof input.number === "number"], but I COOKED MY NOODLE ON THIS ONE DANGIT SO YOU GET TO SUFFER TOO!!*****
+    } else {
+      return "*Bark!*bark!*bark!* (What? this isn't a number... try again, please.)";  //***** Doggo(s) mad.*****
+    }
+    if (bone % 3 === 0) {
+      return `${bone} is divisible by three! :D`;
+    } else {
+      return `${bone} is not divisible by three! >:(`;
+    }
+  };
+
+
+
+
+
+// --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+// input: an array of words (strings)
+//output: an array of words that have had their first letters capitalized (still strings)
+// pseudocode: create a function that will take an array as an input, and using the .map method, capitalize the first letter in each string.
+
+// *****pseudo's kinda vague here because I'm pretty sure that's the proper method, but beyond that I don't really know what to do? I know toUpperCase is a thing, gonna start there. also not gonna modify it if I find a better way, for "thought process" purposes*****

```

Comment:
This is perfect and sometimes it happens this way, just be sure to document the process 😁